ratatui-widgets/src/paragraph.rs

```diff
@@ -477,7 +477,11 @@ fn render_line(wrapped: &WrappedLine<'_, '_>, area: Rect, buf: &mut Buffer, y: u
         // Make sure to overwrite any previous character with a space (rather than a zero-width)
         let symbol = if symbol.is_empty() { " " } else { symbol };
         let position = Position::new(area.left() + x, area.top() + y);
-        buf[position].set_symbol(symbol).set_style(*style);
+        if let Some(cell) = buf.cell_mut(position) {
+            cell.set_symbol(symbol).set_style(*style);
+        } else {
+            break;
+        }
```

Comment:
Another approach that might also be worth considering for this is pre-trimming the area function in the render function. (e.g. `let area = buf.area.intersection(area);` But I think that would change how paragraphs that are partially out of the area are rendered, which may not be desirable. Longer note about this in the test below.